src/main/java/io/stargate/sgv2/jsonapi/service/embedding/DataVectorizer.java

```diff
@@ -102,12 +106,28 @@ public Uni<Boolean> vectorize(List<JsonNode> documents) {
             .onItem()
             .transform(
                 vectorData -> {
+                  // check if we get back the same number of vectors that we asked for
+                  if (vectorData.size() != vectorizeTexts.size()) {
+                    throw EMBEDDING_PROVIDER_INVALID_RESPONSE.toApiException(
+                        "Embedding provider '%s' cannot return the correct number of vectors. Expect: '%s'. Actual: '%s'",
```

Comment:
Can this be rephared as `Embedding provider '%s' didn't return the expected number of embeddings`?